framework/src/meshgenerators/XYZDelaunayGenerator.C

```diff
@@ -340,36 +420,95 @@ XYZDelaunayGenerator::generate()
   auto & main_subdomain_map = mesh->set_subdomain_name_map();
   for (auto hole_i : index_range(_hole_ptrs))
   {
-    if (hole_i < _stitch_holes.size() && _stitch_holes[hole_i])
-    {
-      UnstructuredMesh & hole_mesh = dynamic_cast<UnstructuredMesh &>(**_hole_ptrs[hole_i]);
-      auto & hole_boundary_info = hole_mesh.get_boundary_info();
-
-      // Our algorithm here requires a serialized Mesh.  To avoid
-      // redundant serialization and deserialization (libMesh
-      // MeshedHole and stitch_meshes still also require
-      // serialization) we'll do the serialization up front.
-      libMesh::MeshSerializer serial_hole(hole_mesh);
-
-      // We'll look for any sides that match between the hole mesh and
-      // the newly triangulated mesh, and apply bcids accordingly.
-      for (auto elem : hole_mesh.element_ptr_range())
-        for (auto s : make_range(elem->n_sides()))
-          if (!elem->neighbor_ptr(s))
-          {
-            auto points = sorted_point_tuple(*elem, s);
-            auto it = mesh_faces.find(points);
+    UnstructuredMesh & hole_mesh = dynamic_cast<UnstructuredMesh &>(**_hole_ptrs[hole_i]);
+    auto & hole_boundary_info = hole_mesh.get_boundary_info();
+
+    // Our algorithm here requires a serialized Mesh.  To avoid
+    // redundant serialization and deserialization (libMesh
+    // MeshedHole and stitch_meshes still also require
+    // serialization) we'll do the serialization up front.
+    libMesh::MeshSerializer serial_hole(hole_mesh);
+
+    // We'll look for any sides that match between the hole mesh and
+    // the newly triangulated mesh, and apply bcids accordingly.
+    for (auto elem : hole_mesh.element_ptr_range())
+      for (auto s : make_range(elem->n_sides()))
+        if (!elem->neighbor_ptr(s))
+        {
+          auto points = sorted_point_tuple(*elem, s);
+          auto it = mesh_faces.find(points);
 
-            // I'd love to assert that we don't have any missing
-            // matches, but our holes might themselves have holes
-            if (it != mesh_faces.end())
+          // I'd love to assert that we don't have any missing
+          // matches, but our holes might themselves have holes
+          if (it != mesh_faces.end())
+          {
+            auto [main_elem, main_side] = it->second.first;
+            if (hole_i < _stitch_holes.size() && _stitch_holes[hole_i])
```

Comment:
I m not sure about this. So some holes among _hole_ptrs dont stitch and the stitch_holes might not be big enough?

I feel like they should be the same size, and some true, others false

Reply:
use `(_stitch_holes.size() && _stitch_holes[hole_i])` as suggested